### PR TITLE
Refactoring testing user screens part 3

### PR DESCRIPTION
**Pull request by irina-baeva**
Testing of user screens is not finished yet, creating PR so not to make it big.
* added more tests on Auth Process for login and signup screen
* continued with refactoring of context logic (WIP) and long parameters (to address comment from consulting).

### CODE REVIEW

File: src/state/authentication/authentication.reducer.js [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
-    case SIGNUP_ERROR:
-    case LOGIN_ERROR:
+    case AUTH_ERROR:
       return {
         ...state,
         errorMessage: payload,
       };
-    case SIGNUP_SUCCESS:
-    case LOGIN_SUCCESS:
+    case LOGIN_SIGNUP_SUCCESS:
```
Comment (thdvbr) [inline src/state/authentication/authentication.reducer.js:25]: can this also be just AUTH_SUCCESS like u combined the error types ?

**Review by thdvbr — APPROVED** [review]

File: src/components/form-fields/ControlledTextInput.component.js [hunk 2]
```diff
@@ -8,56 +8,44 @@ const ErrorText = styled.Text`
...
           <Input
-            placeholder={placeholder}
-            autoCapitalize={autoCapitalize}
-            autoCorrect={autoCorrect}
-            placeholderTextColor={placeholderTextColor}
-            returnKeyType={returnKeyType}
-            blurOnSubmit={blurOnSubmit}
-            textContentType={textContentType}
             onChangeText={onChange}
+            returnKeyType={returnKeyType || 'next'}
```
[the remark this answers is not in the document]
Comment (irina-baeva) [reply src/components/form-fields/ControlledTextInput.component.js:43]: yes, this actually was my plan to do for all props, but then lost of props was still big

File: src/components/form-fields/ControlledTextInput.component.js [hunk 3]
```diff
@@ -8,56 +8,44 @@ const ErrorText = styled.Text`
...
         name={name}
-        defaultValue={defaultValue}
+        defaultValue={defaultValue || ''}
         control={control}
         onFocus={() => {
           inputRef.current.focus();
         }}
         render={({ onChange, value }) => (
           <Input
-            placeholder={placeholder}
```
[the remark this answers is not in the document]
Comment (irina-baeva) [reply src/components/form-fields/ControlledTextInput.component.js:44]: this one is used from   {...props}

File: src/components/form-fields/ControlledTextInput.component.js [hunk 4]
```diff
@@ -8,56 +8,44 @@ const ErrorText = styled.Text`
...
-  textContentType = 'oneTimeCode',
-  style,
-  leftIcon,
-  leftIconContainerStyle,
-  autoCompleteType = 'off',
-  keyboardType,
   errors,
   errorStyle,
+  inputRef,
+  ...props
```
[the remark this answers is not in the document]
Comment (irina-baeva) [reply src/components/form-fields/ControlledTextInput.component.js:29]: yes any props, leftIcon is in ...props, i left only props that used differently and could not be used in {...props}

File: src/state/authentication/authentication.reducer.js (hunk 1 above)
Comment (irina-baeva) [reply src/state/authentication/authentication.reducer.js:25]: i tried but could not solve it, cuz in AUTH_SUCCESS
```
 case AUTH_SUCCESS:
      return {
        ...state,
        token: payload.token,
        username: payload.username,
      };
```
but in login signup 

```

    case LOGIN_SIGNUP_SUCCESS:
      return {
        ...state,
        token: payload.token,
        username: payload.user.username,
        errorMessage: '',
      };
```

sp i did not know how to pass  username: payload.user.username

Comment (irina-baeva) [reply src/state/authentication/authentication.reducer.js:25]: but actually i just can solve it in action creators hmmm i will try in other branch

**Pull request merged**